zipkin-cassandra-core/src/main/java/org/twitter/zipkin/storage/cassandra/Repository.java

```diff
@@ -848,7 +831,7 @@ private int durationIndexBucket(long ts) {
                 .setString("service_name", serviceName)
                 .setString("span_name", spanName)
                 .setInt("bucket", durationIndexBucket(timestamp))
-                .setLong("ts", timestamp)
+                .setBytesUnsafe("ts", serializeTs(timestamp))
```

Comment:
ps we need this because the current datastax client over-eagerly wants a j.u.Date type. So it complains if you serialize as long until you can control this in v3.